### PR TITLE
#161967120 Setup and deploy base application to Heroku

### PR DESCRIPTION
**Pull request by jillo-abdullahi**
**What does this PR do?**

Sets up and deploys base application to Heroku.

**Description of tasks to be completed**

1. Add a `Procfile` for Heroku dyno.
2. Configure `authors/settings.py` to set properties for Debug, Allowed Hosts and Static files.
3. Configure `WhiteNoise` for handling of static files
4. Migrate database after pushing the app to Heroku.
5. Create a superuser from heroku bash to check if the migrations were successful.

**How should this be manually tested?**

- Go to the app [link](https://ah-codetitans.herokuapp.com/admin) on Heroku

- Use the following admin credentials to login
`email`: `admin@andela.com`
`password`: `codetitans32`

- Test creation of a new user using Postman from this url:
`POST`: `https://ah-codetitans.herokuapp.com/api/users`

**Any background contexts you want to provide?**
The url [index](https://ah-codetitans.herokuapp.com/) will lead to a 404 because there are no index urls configured on the app just yet.

**What are the relevant Pivotal Tracker stories?**
[#161967120](https://www.pivotaltracker.com/story/show/161967120)

**Checklist**

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [x] I have made corresponding changes to the documentation

- [x] My changes generate no new warnings

**Screenshots**

screenshot 2018-11-29 at 12 16 35

screenshot 2018-11-29 at 11 56 36

### CODE REVIEW

**Review by mutoromike — CHANGES_REQUESTED** [review]
Kindly add a checklist on your PR.

File: authors/settings.py [hunk 1]
```diff
@@ -15,16 +15,18 @@
...
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = '7pgozr2jn7zs_o%i8id6=rddie!*0f0qy3$oy$(8231i^4*@u3'
+SECRET_KEY = os.getenv('SECRET_KEY')
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = True
+DEBUG = False
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ['0.0.0.0', 'localhost','ah-codetitans.herokuapp.com']
```
Comment (bettblake08) [inline authors/settings.py]: Let the list elements hang/stack on top of each other.

File: authors/staticfiles/main.css [hunk 2]
```diff
@@ -0,0 +1,3 @@
+body {
+    color: black
+}
```
Comment (Bryan-Cee) [inline authors/staticfiles/main.css]: Add a new line here to.

File: .travis.yml [hunk 3]
```diff
@@ -31,4 +31,3 @@ after_success:
 # slack notifications configuration
 notifications:
     slack: andela:LTXqtmTsixouhMtsFDOPnaY5
-
```
Comment (Bryan-Cee) [inline .travis.yml:34]: Deleting this line will not affect the functionality, I suggest you delete it and ensure you leave a newline at the end for the EOL.

File: authors/wsgi.py [hunk 4]
```diff
@@ -13,4 +13,4 @@
 
 os.environ.setdefault("DJANGO_SETTINGS_MODULE", "authors.settings")
 
-application = get_wsgi_application()
+application = get_wsgi_application()
```
Comment (Bryan-Cee) [inline authors/wsgi.py]: Add a new line to this file.

**Review by Bryan-Cee — CHANGES_REQUESTED** [review]
Please make the changes.

**Review by bettblake08 — CHANGES_REQUESTED** [review]

File: authors/settings.py (hunk 1 above)
Comment (jillo-abdullahi) [reply authors/settings.py]: Thanks @bettblake08 This is done.

File: authors/wsgi.py (hunk 4 above)
Comment (jillo-abdullahi) [reply authors/wsgi.py]: Done @Bryan-Cee

File: authors/staticfiles/main.css (hunk 2 above)
Comment (jillo-abdullahi) [reply authors/staticfiles/main.css]: New line added @Bryan-Cee Thanks.

File: .travis.yml (hunk 3 above)
Comment (jillo-abdullahi) [reply .travis.yml:34]: Done! @Bryan-Cee Thanks

**Review by Bryan-Cee — APPROVED** [review]

**Comment by jillo-abdullahi** [comment]
@mutoromike My apologies. Forgot to add the checklist. It's been added now!

**Review by Darius-Ndubi — APPROVED** [review]
👍

**Review by bettblake08 — APPROVED** [review]
👍

**Review by mutoromike — APPROVED** [review]

**Pull request merged**